modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/Slice2DIndex.java

```diff
@@ -159,14 +175,17 @@ public synchronized Slice2DIndex getSlice2DIndex( int imageIndex ) throws IOExce
             long endDataPosition = raf.readLong();
 
             raf.seek(dataPosition);
-            int tIndex = raf.readInt();
-            int zIndex = raf.readInt();
+            int dimensions = raf.readInt();
```

Comment:
I added a commit that supports backwards compatibility. The presence of a negative number indicates that there is a variable dimension size.